poolmgr/gp.go

```diff
@@ -297,9 +316,23 @@ func (gp *GenericPool) specializePod(pod *v1.Pod, metadata *api.ObjectMeta) erro
 	// tell fetcher to get the function.
 	fetcherUrl := gp.getFetcherUrl(podIP)
 	log.Printf("[%v] calling fetcher to copy function", metadata.Name)
-	err := fetcherClient.DoFetchRequest(fetcherUrl, &fetcher.FetchRequest{
+
```

Comment:
So, in general, we have to be very very careful adding anything to this function, because it slows down the cold-start. E.g. I think we should cache the package metadata for functions, that way we don't have to hit the k8s api in this function. However, that caching is a bit of a bigger change, and I'm ok with doing it later.